lib/util/apply-disable-directives.js

```diff
@@ -68,22 +68,38 @@ module.exports = options => {
 
     // enabledRules is only used when there is an active global /* eslint-disable */ comment.
     const enabledRules = new Set();
+    const lineDisabledRules = new Map();
 
     for (const problem of options.problems) {
+        let tempDisabledByLine = new Set();
+        let directive = {};
+        const lineDisabledRule = lineDisabledRules.get(problem.line);
```

Comment:
Why are `enable-line` directives needed if we are filtering by line here anyway?

Reply:
Because sometimes there would be no code on a disabled line and then next line would have code with violations. That would only for some unique scenarios like this one https://github.com/eslint/eslint/blob/22bf943548bc83553111814f1cddccd223b9a35f/tests/lib/util/apply-disable-directives.js#L352